pkg/controller/resourcequota/resource_quota_controller_test.go

```diff
@@ -78,35 +92,195 @@ func newGenericLister(groupResource schema.GroupResource, items []runtime.Object
 	return cache.NewGenericLister(store, groupResource)
 }
 
+type testRESTMapper struct {
+	meta.RESTMapper
+}
+
+func (_ *testRESTMapper) Reset() {}
+
+// fakeAction records information about requests to aid in testing.
+type fakeAction struct {
+	method string
+	path   string
+	query  string
+}
+
+// String returns method=path to aid in testing
+func (f *fakeAction) String() string {
+	return strings.Join([]string{f.method, f.path}, "=")
+}
+
+type FakeResponse struct {
+	statusCode int
+	content    []byte
+}
+
+// fakeActionHandler holds a list of fakeActions received
+type fakeActionHandler struct {
+	// statusCode and content returned by this handler for different method + path.
+	response map[string]FakeResponse
+
+	lock    sync.Mutex
+	actions []fakeAction
+}
+
+// ServeHTTP logs the action that occurred and always returns the associated status code
+func (f *fakeActionHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
```

Comment:
this is a lot of mock I didn't expect to see.

Reply:
the listWatch test doesn't add anything here. That certainly can go, removing some of the mock code.

Reply:
Removed the listWatch test, along with some others. Weren't really adding too much. This also removed most of the mock code.

Please see this commit: https://github.com/kubernetes/kubernetes/pull/64201/commits/9247d036326d3f1e17644ce50b2b4c5e65b0101f. If remove those tests seems fine, I'll squash.